File: src/main/java/com/clinic/infrastructure/notifications/NoOpAppointmentNotifier.java
package com.clinic.infrastructure.notifications;

import com.clinic.domain.entities.Appointment;
import com.clinic.domain.ports.AppointmentNotifier;

/** No-op implementation used when ACS is not configured (ACS_ENDPOINT is absent). */
public class NoOpAppointmentNotifier implements AppointmentNotifier {
  @Override
  public void notifyCompleted(Appointment appointment) {}

  @Override
  public void notifyCancelled(Appointment appointment) {}

  @Override
  public void notifyRescheduled(Appointment old, Appointment newAppointment) {}
}

File: clinic/infrastructure/notifications/no_op_appointment_notifier.py
"""No-op implementation used when ACS is not configured (ACS_ENDPOINT is absent)."""

from __future__ import annotations

from clinic.domain.entities.appointment import Appointment


class NoOpAppointmentNotifier:
    def notify_completed(self, appointment: Appointment) -> None:
        pass

    def notify_cancelled(self, appointment: Appointment) -> None:
        pass

    def notify_rescheduled(self, old: Appointment, new_appointment: Appointment) -> None:
        pass
